Module not found: Error: Can't resolve 'react-dom/server' on develop when using gatsby-plugin-typography and gatsby-plugin-preact

### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

"Module not found: Error: Can't resolve 'react-dom/server'" at running `gatsby develop` when adding `gatsby-plugin-preact` to a project using `gatsby-plugin-typography`

- gatsby-plugin-typography -> gatsby develop -> No Error
- gatsby-plugin-typography + gatsby-plugin-preact -> gatsby develop -> Error!
- gatsby-plugin-typography + gatsby-plugin-preact -> gatsby build -> No Error

### Reproduction Link

https://github.com/tamaosa/bug-repro-typography-preact

### Steps to Reproduce

1. clone repository
2. gatsby develop



### Expected Result

No error on develop

### Actual Result

The following error occurs.

```shell
 ERROR #98124  WEBPACK

Generating development JavaScript bundle failed

Can't resolve 'react-dom/server' in '/home/user/bug-repro-typography-preact/node_modules/gatsby-plugin-typography'

If you're trying to use a package make sure that 'react-dom/server' is installed. If you're trying to use a local file make sure that the path is correct.

File: node_modules/gatsby-plugin-typography/gatsby-browser.js:9:19

failed Building development bundle - 42.448s
ERROR in ./node_modules/gatsby-plugin-typography/gatsby-browser.js 9:19-46
Module not found: Error: Can't resolve 'react-dom/server' in '/home/user/bug-repro-typography-preact/node_modules/gatsby-plugin-typography'
 @ ./.cache/api-runner-browser-plugins.js 2:10-79
 @ ./.cache/api-runner-browser.js 1:14-53
 @ ./.cache/app.js 9:0-65 26:87-30:1 34:20-29 47:0-14 113:17-26 148:6-15 26:0-30:2

develop compiled with 1 error
```

### Environment

```shell
System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa) 
    CPU: (8) x64 AMD Ryzen 5 2500U with Radeon Vega Mobile Gfx
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Languages:
    Python: 3.9.9 - /home/user/.pyenv/shims/python
  npmPackages:
    gatsby: ^4.6.1 => 4.6.1
    gatsby-plugin-preact: ^6.6.0 => 6.6.0
    gatsby-plugin-typography: ^4.6.0 => 4.6.0
```


### Config Flags

_No response_

--- a/packages/gatsby-plugin-preact/src/gatsby-node.js
+++ b/packages/gatsby-plugin-preact/src/gatsby-node.js
@@ -65,9 +65,7 @@ export function onCreateWebpackConfig({ stage, actions, getConfig }) {
     resolve: {
       alias: {
         react: require.resolve(`preact/compat`).replace(`.js`, extension),
-        "react-dom/server": require
-          .resolve(`preact/compat/server`)
-          .replace(`.js`, extension),
+        "react-dom/server": require.resolve(`preact/compat/server`),
         "react-dom": require.resolve(`preact/compat`).replace(`.js`, extension),
         "react/jsx-runtime": require
           .resolve(`preact/jsx-runtime`)

